Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -419,30 +429,34 @@ public class PBXProjGenerator {
             )
         )
 
-        guard let productType = targetObject.productType,
-            let buildConfigurations = targetObject.buildConfigurationList?.buildConfigurations,
-            let defaultConfigurationName = targetObject.buildConfigurationList?.defaultConfigurationName,
-            let defaultConfiguration = buildConfigurations.first(where: { $0.name == defaultConfigurationName }) ?? buildConfigurations.first else {
+        if let platform = platform {
```

Comment:
I think it's better to split up this method into "generate targetDependency" and others instead of allowing nil platform.
I expect that it will make it unnecessary to [unwrap `Optional` forcibly](https://github.com/yonaskolb/XcodeGen/pull/797/files#diff-2f3c621620623f55e7ff42cc32198a10R706).